osltweaks.php

```diff
@@ -44,6 +44,24 @@ function osltweaks_civicrm_buildForm($formName, &$form) {
         }
       }
     }
+
+    // If contribution page id exist in $settings['member_redirect_contribution_pages'] array
+    if (array_key_exists($form->_id, ($settings['member_redirect_contribution_pages'] ?? array()))) {
+      // Get current contact id
+      $currentContactId = CRM_Core_Session::singleton()->getLoggedInContactID();
+      // Get memberships count base on the current contact id
+      $memberships = civicrm_api3('Membership', 'getcount', [
+        'contact_id' => $currentContactId,
+      ]);
+
+      // Redirect if there are results on memberships api
+      if ($memberships) {
+        // Parse current URL to get the current q
+        parse_str(parse_url($form->controller->_entryURL, PHP_URL_QUERY), $parseUrl);
+        $redirect = $settings['member_redirect_contribution_pages'][$form->_id];
+        CRM_Utils_System::redirect(CRM_Utils_System::url($parseUrl['amp;q'], "reset=1&id={$redirect}"));
```

Comment:
@lucky091588 I'm surprised to see `'amp;q'` here instead of just `'q'`. Can you describe why the 'amp;' is necessary?

Reply:
@twomice I forgot to decode it. See fix in the next commit.